core/src/main/java/nl/nn/adapterframework/scheduler/JobDef.java

```diff
@@ -696,26 +692,75 @@ private void cleanupDatabase(IbisManager ibisManager) {
 				}
 			}
 		}
+		return messageLogs;
+	}
 
-		for (MessageLogObject mlo: messageLogs) {
-			setJmsRealm(mlo.getJmsRealmName());
-			DirectQuerySender qs;
-			qs = (DirectQuerySender)ibisManager.getIbisContext().createBeanAutowireByName(DirectQuerySender.class);
-			qs.setJmsRealm(mlo.getJmsRealmName());
-			String deleteQuery;
-			if (qs.getDatabaseType() == Dbms.MSSQL) {
-				deleteQuery = "DELETE FROM " + mlo.getTableName() + " WHERE " + mlo.getKeyField() + " IN (SELECT " + mlo.getKeyField() + " FROM " + mlo.getTableName()
-						+ " WITH (readpast) WHERE " + mlo.getTypeField() + " IN ('" + IMessageBrowser.StorageType.MESSAGELOG_PIPE.getCode() + "','" + IMessageBrowser.StorageType.MESSAGELOG_RECEIVER.getCode()
-						+ "') AND " + mlo.getExpiryDateField() + " < "+qs.getDbmsSupport().getDatetimeLiteral(date)+")";
+	private void cleanupDatabase(IbisManager ibisManager) {
+		Date date = new Date();
+
+		List<String> datasourceNames = getAllLockerDatasourceNames(ibisManager);
+
+		for (Iterator<String> iter = datasourceNames.iterator(); iter.hasNext();) {
+			String datasourceName = iter.next();
+			DirectQuerySender qs = null;
+			String deleteQuery = null;
+			try {
+				qs = ibisManager.getIbisContext().createBeanAutowireByName(DirectQuerySender.class);
+				qs.setDatasourceName(datasourceName);
+				qs.setName("executeQueryJob");
+				qs.setQueryType("other");
+				qs.setTimeout(getQueryTimeout());
+				qs.configure(true);
+				qs.open();
+
+				deleteQuery = "DELETE FROM IBISLOCK WHERE EXPIRYDATE < "+qs.getDbmsSupport().getDatetimeLiteral(date);
+				Message result = qs.sendMessage(new Message(deleteQuery), null);
+				log.info("result [" + result + "]");
+			} catch (Exception e) {
+				String msg = "error while executing query ["+deleteQuery+"] (as part of scheduled job execution): " + e.getMessage();
+				getMessageKeeper().add(msg, MessageKeeperLevel.ERROR);
+				log.error(getLogPrefix()+msg);
+			} finally {
+				if(qs != null) {
+					qs.close();
+				}
 			}
-			else {
-				deleteQuery = "DELETE FROM " + mlo.getTableName() 
-					+ " WHERE " + mlo.getTypeField() + " IN ('" + IMessageBrowser.StorageType.MESSAGELOG_PIPE.getCode() + "','" + IMessageBrowser.StorageType.MESSAGELOG_RECEIVER.getCode() + "') AND " + mlo.getExpiryDateField() + " < "+qs.getDbmsSupport().getDatetimeLiteral(date);
+		}
+
+		List<MessageLogObject> messageLogs = getAllMessageLogsAndErrorstorages(ibisManager);
+
+		for (MessageLogObject mlo: messageLogs) {
+			DirectQuerySender qs = null;
+			String deleteQuery = null;
+			try {
+				qs = ibisManager.getIbisContext().createBeanAutowireByName(DirectQuerySender.class);
+				qs.setDatasourceName(mlo.getDatasourceName());
+				qs.setName("executeQueryJob");
+				qs.setQueryType("other");
+				qs.setTimeout(getQueryTimeout());
+				qs.configure(true);
+				qs.open();
+
+				if (qs.getDatabaseType() == Dbms.MSSQL) {
```

Comment:
Hier moet niet een specifieke test op Dbms.MSSQL zijn, maar moet er gebruik gemaakt worden van generieke functies in dbmsSupport. Zal ik wel doen na mijn vakantie.
Ik wil dat wat nu (geloof ik) heet prepareSessionForDirtyRead() hernoemen in prepareSessionForNonLockingRead(), en de method prepareQueryForWorkqueuePeeking() in prepareQueryForNonLockingRead(). 
Die moeten dan hier ook gebruikt worden.